Sources/X509/CryptographicMessageSyntax/CMSOperations.swift

```diff
@@ -23,15 +28,82 @@ public enum CMS {
         signatureAlgorithm: Certificate.SignatureAlgorithm,
         additionalIntermediateCertificates: [Certificate] = [],
         certificate: Certificate,
-        privateKey: Certificate.PrivateKey
+        privateKey: Certificate.PrivateKey,
+        includeSigningTime: Bool = false
     ) throws -> [UInt8] {
-        let signature = try privateKey.sign(bytes: bytes, signatureAlgorithm: signatureAlgorithm)
-        return try sign(
+        var signature: Certificate.Signature
+        var completeSignedAttrs: [CMSAttribute]? = nil
+        if includeSigningTime {
+            var signedAttrs: [CMSAttribute] = []
+            // As specified in RFC 5652 section 11 when including signedAttrs we need to include a minimum of:
+            // 1. content-type
+            // 2. message-digest
+
+            // add content-type signedAttr cms data
+            let contentTypeVal = try ASN1Any(erasing: ASN1ObjectIdentifier.cmsData)
+            let contentTypeAttribute = CMSAttribute(attrType: .contentType, attrValues: [contentTypeVal])
+            signedAttrs.append(contentTypeAttribute)
+
+            // add message-digest sha256 of provided content bytes
+            try SHA256.hash(data: bytes).withUnsafeBytes { bufferPointer in
+                let messageDigest = ASN1OctetString(contentBytes: ArraySlice(bufferPointer))
+                let messageDigestVal = try ASN1Any(erasing: messageDigest)
+                let messageDigestAttribute = CMSAttribute(attrType: .messageDigest, attrValues: [messageDigestVal])
+                signedAttrs.append(messageDigestAttribute)
+            }
+
+            // add signing time utc time in 'YYMMDDHHMMSSZ' format as specificed in `UTCTime`
+            let desiredComponents: Set = [
+                Calendar.Component.year,
+                Calendar.Component.month,
+                Calendar.Component.day,
+                Calendar.Component.hour,
+                Calendar.Component.minute,
+                Calendar.Component.second,
+            ]
+            guard let timeZone = TimeZone(identifier: "UTC") else {
+                throw CMSSigningError.noTimeZone
+            }
+            var calendar = Calendar(identifier: .gregorian)
+            calendar.timeZone = timeZone
+            let nowComponents = calendar.dateComponents(desiredComponents, from: Date())
```

Comment:
We should avoid this. We have a `Date.utcTime` transformation that already exists.